lib/coverband/collectors/coverage.rb

```diff
@@ -17,27 +17,25 @@ def self.instance
       def reset_instance
         @project_directory = File.expand_path(Coverband.configuration.root)
         @file_line_usage = {}
-        @ignored_files = Set.new
         @ignore_patterns = Coverband.configuration.ignore + ['internal:prelude', 'schema.rb']
         @reporting_frequency = Coverband.configuration.reporting_frequency
         @store = Coverband.configuration.store
         @verbose  = Coverband.configuration.verbose
         @logger   = Coverband.configuration.logger
         @current_thread = Thread.current
         @test_env = Coverband.configuration.test_env
-        @background_reporting_enabled = Coverband.configuration.background_reporting_enabled
+        @@previous_results = nil
         Thread.current[:coverband_instance] = nil
         self
       end
 
       def report_coverage(force_report = false)
-        return Background.start if @background_reporting_enabled
         return if !ready_to_report? && !force_report
-
         unless @store
           @logger.debug 'no store set, no-op'
           return
```

Comment:
Do we want to raise an error here? Could be confusing why nothing is working. Another approach could be to throw an error on Coverband.configure. That way we never even get here.

Reply:
One other thought, is a nil @store even possible? 
https://github.com/danmayer/coverband/pull/146/files#diff-1f4b05f150269d04438736256c27b0a8R62

Reply:
Yeah I had thought a bit about that... I guess it would be nice to push this into configure... basically if you don't set a store and the default one fails for some reason raise an error. 

The issue would be if the error only happens in prod it could be a scary experience... I have tried to make sure Coverband is always safe and doesn't hurt prod when it isn't working... Thoughts?

Reply:
Looks like we are already rescuing all errors anyway?

https://github.com/danmayer/coverband/pull/146/files#diff-181f764c69748a3ceb0cc64e4a37ebaeR43

Reply:
yeah true... I change to a raise which gets caught and logged, I don't think configure would get caught since that would happen outside of this rescue. Anyways should make it a bit more obvious something is wrong.